packages/core/src/utils/queuedStorage/createQueuedStorage.ts

```diff
@@ -0,0 +1,165 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import { DATABASE_NAME, STORE_NAME } from './constants';
+import { getAddItemBytesSize } from './getAddItemBytesSize';
+import { AddItemWithBytesSize, QueuedItem, QueuedStorage } from './types';
+
+export const createQueuedStorage = (): QueuedStorage => {
+	let currentBytesSize = 0;
+	let error: DOMException | undefined;
+
+	const openDBPromise = new Promise<IDBDatabase>((resolve, reject) => {
+		const indexDB = window.indexedDB;
```

Comment:
Nit:
```suggestion
		const indexedDB = window.indexedDB;
```
Or.. is it possible to destructure it directly from `window`?

Reply:
I thought I fixed all the names 😵‍💫